src/shared/user-record.c

```diff
@@ -217,12 +217,21 @@ static int json_dispatch_gecos(const char *name, JsonVariant *variant, JsonDispa
                 return json_log(variant, flags, SYNTHETIC_ERRNO(EINVAL), "JSON field '%s' is not a string.", strna(name));
 
         n = json_variant_string(variant);
-        if (!valid_gecos(n))
-                return json_log(variant, flags, SYNTHETIC_ERRNO(EINVAL), "JSON field '%s' is not a valid GECOS compatible real name.", strna(name));
+        if (!valid_gecos(n)) {
+                _cleanup_free_ char *m = NULL;
 
-        r = free_and_strdup(s, n);
-        if (r < 0)
-                return json_log(variant, flags, r, "Failed to allocate string: %m");
+                json_log(variant, flags|JSON_DEBUG, SYNTHETIC_ERRNO(EINVAL), "JSON field '%s' is not a valid GECOS compatible real name, mangling.", strna(name));
+
+                m = mangle_gecos(n);
+                if (!m)
+                        return log_oom();
+
+                free_and_replace(*s, m);
+        } else {
```

Comment:
Hmm, the order is "valid-invalid" above "invalid-valid" here. My OCD is triggered ;)

Reply:
Now you triggered my OCD, too.

Force pushed a new version. only change: the two branches are swapped. No other changes.